sections/cart-drawer.liquid

```diff
@@ -0,0 +1,334 @@
+<script src="{{ 'cart.js' | asset_url }}" defer="defer"></script>
+
+<style>
+  .drawer {
+    visibility: hidden;
+  }
+</style>
+
+<cart-drawer class="drawer{% if cart == empty %} is-empty{% endif %}">
+  <div id="CartDrawer" class="cart-drawer">
+    <div id="CartDrawer-Overlay"class="cart-drawer__overlay"></div>
+    <div class="drawer__inner" role="dialog" aria-modal="true" aria-label="{{ 'sections.cart.title' | t }}" tabindex="-1">
+       {%- if cart == empty -%}
+        <div class="drawer__inner-empty">
+          <div class="cart-drawer__warnings center{% if settings.cart_drawer_collection != blank %} cart-drawer__warnings--has-collection{% endif %}">
+            <div class="cart-drawer__empty-content">
+              <h2 class="cart__empty-text">{{ 'sections.cart.empty' | t }}</h2>
+              <button class="drawer__close" type="button" onclick="this.closest('cart-drawer').close()" aria-label="{{ 'accessibility.close' | t }}">{% render 'icon-close' %}</button>
+              <a href="{{ routes.all_products_collection_url }}" class="button">
+                {{ 'general.continue_shopping' | t }}
+              </a>
```

Comment:
So, currently, this button redirects the user to `collections/all`. But let's say I'm browsing a product, I open the cart drawer and it's empty. Should this button:
1. Redirect me to the `collections/all` page, like in the Cart Page **OR**
2. Simply close the drawer, like in the Cart Popup?

I think **Option 1** makes sense in the context of the cart page, where you don't usually have anything to shop on that page. But in a cart drawer that opens as an overlay, I'm not fully convinced of this. I think **Option 2** kinda feels more natural to me - just acting as an additional close button.

This is not really a suggestion as I'm not fully convinced of the best UX. But I just wanted to share this to get more context behind this decision and hear thoughts from others.

### Cart Drawer

https://user-images.githubusercontent.com/37168033/166566890-8352c51f-ec0f-49e6-b1f3-b8550a14836d.mp4

### Popup notification

https://user-images.githubusercontent.com/37168033/166567031-235ce782-32cf-429d-80f1-6622edc1065c.mp4

Reply:
Just to add: I remember that during one of our accessibility testing, the user told us that they couldn't easily tell the difference between `Close` and `Continue shopping` buttons in the cart popup and that it was not immediately obvious that they performed the same action.

So I'm not sure replicating that behaviour here is the solution. But just wanted to bring this to discussion, as we currently have different behaviours for the same label in these two different cart types.

Reply:
> 1. Redirect me to the `collections/all` page, like in the Cart Page **OR**
> 2. Simply close the drawer, like in the Cart Popup?
> 
> I think **Option 1** makes sense in the context of the cart page, where you don't usually have anything to shop on that page. But in a cart drawer that opens as an overlay, I'm not fully convinced of this. I think **Option 2** kinda feels more natural to me - just acting as an additional close button without interrupting the user's shopping flow.

Thanks for bringing this up, Lucas. Here's my take on the empty state "Continue shopping" button:

I agree that the "Continue shopping" button should simply close the drawer as it does feel more natural to me as well. There could be instances where users are browsing products (on the 5th or 6th collection page for example) and in the current state, if they click Continue shopping, they'd be redirected to the another collection of the same collection but on the first page. They could just hit the back button but that's probably not ideal and also leads to my next point.

Having worked on the other side of the fence (theme support), one customization request we often got was adding a setting to allow merchants to use their own/custom `link` for the continue shopping button. The same setting could then be added to the page and popup cart types since they also include "Continue shopping" - _might_ be something we could add in the future perhaps.